File: CS/ReadOnlyTabStopRemover.Module/Detail.cs
using System;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;

namespace ReadOnlyTabStopRemover.Module
{
    [NavigationItem(true)]
    public class Detail : BaseObject
    {
        public Detail(Session session) : base (session)
        {

        }

        #region DocumentNumber

        private string _property;

        [Size(10)]
        public string DocumentNumber
        {
            get
            {
                return _property;
            }
            set
            {
                SetPropertyValue<string>("DocumentNumber", ref _property, value);
            }
        }

        #endregion

        #region DocumentNumberRO

        public string DocumentNumberRO
        {
            get
            {
                return this.DocumentNumber;
            }
        }

        #endregion

        #region DocumentReference

        private string _documentReference;

        [Size(10)]
        public string DocumentReference
        {
            get
            {
                return _documentReference;
            }
            set
            {
                SetPropertyValue<string>("DocumentReference", ref _documentReference, value);
            }
        }

        #endregion

        #region DocumentReferenceRO

        public string DocumentReferenceRO
        {
            get
            {
                return this.DocumentReference;
            }
        }

        #endregion

        #region DocumentDate

        private DateTime _documentDate;

        public DateTime DocumentDate
        {
            get
            {
                return _documentDate;
            }
            set
            {
                SetPropertyValue<DateTime>("DocumentDate", ref _documentDate, value);
            }
        }

        #endregion

        #region DocumentDateRO

        public DateTime DocumentDateRO
        {
            get
            {
                return this.DocumentDate;
            }
        }

        #endregion
    }
}

File: VB/ReadOnlyTabStopRemover.Module/Detail.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo

Namespace ReadOnlyTabStopRemover.Module
	<NavigationItem(True)> _
	Public Class Detail
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)

		End Sub

		#Region "DocumentNumber"

		Private _property As String

		<Size(10)> _
		Public Property DocumentNumber() As String
			Get
				Return _property
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("DocumentNumber", _property, value)
			End Set
		End Property

		#End Region

		#Region "DocumentNumberRO"

		Public ReadOnly Property DocumentNumberRO() As String
			Get
				Return Me.DocumentNumber
			End Get
		End Property

		#End Region

		#Region "DocumentReference"

		Private _documentReference As String

		<Size(10)> _
		Public Property DocumentReference() As String
			Get
				Return _documentReference
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("DocumentReference", _documentReference, value)
			End Set
		End Property

		#End Region

		#Region "DocumentReferenceRO"

		Public ReadOnly Property DocumentReferenceRO() As String
			Get
				Return Me.DocumentReference
			End Get
		End Property

		#End Region

		#Region "DocumentDate"

		Private _documentDate As DateTime

		Public Property DocumentDate() As DateTime
			Get
				Return _documentDate
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue(Of DateTime)("DocumentDate", _documentDate, value)
			End Set
		End Property

		#End Region

		#Region "DocumentDateRO"

		Public ReadOnly Property DocumentDateRO() As DateTime
			Get
				Return Me.DocumentDate
			End Get
		End Property

		#End Region
	End Class
End Namespace
